src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -622,14 +557,14 @@ private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
                     .FirstOrDefault(section =>
                         string.Equals(
                             section.Key,
-                            _microsoftFeatureManagementSchemaEnabled ? 
-                                MicrosoftFeatureManagementFields.FeatureManagementSectionName : 
-                                ConfigurationFields.FeatureManagementSectionName,
+                            _microsoftFeatureManagementSchemaEnabled ?
+                                MicrosoftFeatureManagementFields.FeatureManagementSectionName :
+                                DotnetFeatureManagementFields.FeatureManagementSectionName,
                             StringComparison.OrdinalIgnoreCase));
 
             if (featureManagementConfigurationSection == null)
             {
-                if (RootConfigurationFallbackEnabled && !_microsoftFeatureManagementSchemaEnabled)
```

Comment:
Can you explain why this condition is removed? I think I missed why it was added originally

Reply:
We have this method: `IFeatureManagementBuilder AddFeatureManagement(this IServiceCollection services, IConfiguration configuration)` where you can pass a section of configuration and the configurationFeatureDefintionProvider will fall back to the root of configuration if feature management section cannot be found.

We also mention it in our [public doc](https://learn.microsoft.com/en-us/azure/azure-app-configuration/use-feature-flags-dotnet-core#set-up-feature-management)

We are promoting Microsoft schema as primary schema. I think the primary schema should be supported for this usage.

In #319, I supported this root fall back behavior for Microsoft schema. And we have to use the EnsureInit pattern for it.
At that time, the microsoft schema is still under `FeatureManagement` section.
 
In #370, we decided to put microsoft schema under `feature_management` section. I want to get rid of the EnsureInit pattern, so I talked with Jimmy, asked whether we can discard the configuration root fall back behavior for microsoft schema. Jimmy agreed that. So I removed the EnsureInit pattern and add this check: 
`if (RootConfigurationFallbackEnabled && !_microsoftFeatureManagementSchemaEnabled)`
At that time, .NET schema is the primary schema, so I think it is ok to not support this for the secondary microsoft schema.